viewing_party/party.py

```diff
@@ -1,23 +1,159 @@
 # ------------- WAVE 1 --------------------
 
+from collections import UserString
+from enum import unique
+from re import U
+
+
 def create_movie(title, genre, rating):
-    pass
+    if not(title and genre and rating):
+        return None
+
+    movie = {}
+    
+    movie['title'] = title
+    movie['genre'] = genre
+    movie['rating'] = rating
+
+    return movie 
+
+def add_to_watched(user_data, movie):
+    user_data['watched'].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data['watchlist'].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    movie_watched = None
+
+    for movie in user_data['watchlist']:
+        if movie['title'] == movie_title:
+            movie_watched = movie
+            break
+            
+    if movie_watched:
+        user_data['watched'].append(movie_watched)
+        user_data['watchlist'].remove(movie_watched)
+    
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    total_ratings = 0
+    average_rating = 0.0
+
+    if user_data['watched']:
+        # find the number of movies and add all ratings together
+        for movie in user_data['watched']:
+            # num_movies += 1
+            total_ratings += movie['rating']
+        average_rating = total_ratings / len(user_data['watched'])
+    
+    return average_rating
+
+def get_most_watched_genre(user_data):
+    genre_counts = {}
+    most_watched_genre = None
 
+    if user_data['watched']:
+        # create dictionary with genre as key and frequency of genre as value
+        for movie in user_data["watched"]:
+            if movie['genre'] not in genre_counts:
+                genre_counts[movie['genre']] = 1
+            elif movie['genre'] in genre_counts:
+                genre_counts[movie['genre']] += 1
+        #use max to get the key with highest frequency from dictionary
+        most_watched_genre = max(genre_counts, key=genre_counts.get)

```

Comment:
great use of the max method. If you didn't use the max method what would you do instead?